app/scripts/views/mixins/signup-success-mixin.js

```diff
@@ -7,16 +7,28 @@
 define(function (require, exports, module) {
   'use strict';
 
+  var t = require('views/base').t;
+  var Constants = require('lib/constants');
 
   module.exports = {
     onSignUpSuccess: function (account) {
       var self = this;
       if (account.get('verified')) {
-        // user was pre-verified, notify the broker.
-        return self.invokeBrokerMethod('afterSignIn', account)
-          .then(function () {
-            self.navigate('signup_complete');
-          });
+        if (self.relier.get('preVerifyToken')) {
+          // User was pre-verified, notify the broker.
+          return self.invokeBrokerMethod('afterSignIn', account)
+            .then(function () {
+              self.navigate('signup_complete');
+            });
+        }
+
+        // Account already existed. There was no need to create it,
+        // so we just signed the user in instead.
+        // https://github.com/mozilla/fxa-content-server/issues/2778
+        return self.navigate('settings', {
```

Comment:
Another example of the implicit screen->screen state machine being a PITA.

Blanket redirecting to [`settings` is problematic for the firstrun flow](https://github.com/mozilla/fxa-content-server/blob/77e711232ed9c690b24af4b1c2d9419f41bd73bc/app/scripts/models/auth_brokers/first-run.js#L43-L50).

I would do something like the following for lines 17-31:

```js
return self.invokeBrokerMethod('afterSignIn', account)
  .then(function () {
    if (self.relier.has('preVerifyToken')) {
      return self.navigate('signup_complete');
    } else {
      return self.navigate('settings', {
        success: t(Constants.SIGN_UP_EXISTING_USER_SUCCESS)
      });
    }
  });
```

Let the broker say whether to continue or not, and then redirect to the appropriate screen.